pandas/core/indexes/base.py

```diff
@@ -594,6 +596,40 @@ def values(self):
         """ return the underlying data as an ndarray """
         return self._data.view(np.ndarray)
 
+    @property
+    def _values(self):
+        # type: () -> Union[ExtensionArray, Index]
+        # TODO: remove index types as they become is extension arrays
+        """The best array representation.
+
+        This is an ndarray, ExtensionArray, or Index subclass. This differs
+        from ``_ndarray_values``, which always returns an ndarray.
+
+        Both ``_values`` and ``_ndarray_values`` are consistent between
+        ``Series`` and ``Index``.
+
+        It may differ from the public '.values' method.
+
+        index             | values          | _values     | _ndarray_values |
+        ----------------- | -------------- -| ----------- | --------------- |
+        CategoricalIndex  | Categorical     | Categorical | codes           |
+        DatetimeIndex[tz] | ndarray[M8ns]   | DTI[tz]     | ndarray[M8ns]   |
+
+        For the following, the ``._values`` is currently ``ndarray[object]``,
+        but will soon be an ``ExtensionArray``
+
+        index             | values          | _values      | _ndarray_values |
+        ----------------- | --------------- | ------------ | --------------- |
+        PeriodIndex       | ndarray[object] | ndarray[obj] | ndarray[int]    |
```

Comment:
For the `_values`, shouldn't this be `PeriodIndex` for now? Similar as for DatetimeTZ now has DTI[tz], and so in the future it would become PeriodArray. 

(but not sure where this is currently actually used)

Reply:
No I don't think so, since `Series[Period]._values` returns an ndarray of objects. `Series[datetime-with-TZ]._values` is the only special one, and I'm only changing `Index._values` to match `Series._values` for now.

Reply:
Yes, but that is because we don't have a PeriodBlock that is backed by a PeriodIndex, as is the case for DatetimeTZ, so at the moment you cannot really have Period values in a series. 
I am not saying that this should be the case now, only that I think it might be more logical. But, it might well be that we then first need to add a working "PeriodBlock" (ExtensionBlock backed by PeriodIndex/PeriodArray) to have this work.

Reply:
I'm confused what your proposing / saying then. Is it just that `._values` being a `PeriodIndex` would be more consistent with `._values` returning a `DatetimeIndex`?

What we have now "works", but is internally inconsistent. I'm updating `Index._values` to be consistent with `Series._values`, and changing uses of `.values` / `._values` / `._ndarray_values` to be the correct one for that use.

Reply:
> Is it just that ._values being a PeriodIndex would be more consistent with ._values returning a DatetimeIndex?

Yes. So regardless of what `._values` currently does or doesn't return for Index/Series. Just from looking at your table, returning `PeriodIndex` would make more sense IMO, as this is (at this moment) the closest thing to an array-like that preserves the information. 
But, I don't know what implication it would have for Series ops that `._values` would start returning a PeriodIndex instead of object array of Periods, so it was really just a question, not a request for change.